File: sample/javascript/mainModel.mjs
import BuildBox from './buildBox.mjs';
import { getBoxesFromPly } from './plyUtil.mjs';

(async () => {
  const roomName = '1000';
  const buildBox = new BuildBox(roomName);

  buildBox.setBoxSize(0.5);
  buildBox.setBuildInterval(0.01);

  const plyFileName = 'piyo.ply'

  const boxes = getBoxesFromPly(plyFileName)

  for (const b of boxes) {
    // console.log(b)
    buildBox.createBox(b[0], b[1], b[2], b[3], b[4], b[5], b[6])
  }

  await buildBox.sendData();
  console.log('send data done')
})().catch(error => {
  console.error(error);
});

File: sample/python/main_model.py
from build_box import BuildBox
from ply_util import get_boxes_from_ply

room_name = "1000"
build_box = BuildBox(room_name)

build_box.set_box_size(1)
build_box.set_build_interval(0.01)
build_box.set_node(0, 0, 0, pitch=0, yaw=0, roll=0)

ply_file_name = 'piyo.ply'

boxes = get_boxes_from_ply(ply_file_name)

for b in boxes:
    build_box.create_box(b[0], b[1], b[2], r=b[3], g=b[4], b=b[5], alpha=b[6])

build_box.send_data()
